### PR TITLE
Operator API

### PR DESCRIPTION
**Pull request by Spikhalskiy**

### CODE REVIEW

File: client/frontend/client.go [hunk 1]
```diff
@@ -63,6 +63,11 @@ func NewClient(
 	}
 }
 
+func (c *clientImpl) GetWorkflowExecutionHistoryReverse(ctx context.Context, in *workflowservice.GetWorkflowExecutionHistoryReverseRequest, opts ...grpc.CallOption) (*workflowservice.GetWorkflowExecutionHistoryReverseResponse, error) {
+	//TODO remove
+	panic("implement me")
+}
+
```
Comment (yiminc) [inline client/frontend/client.go:70]: please rebase, the reverse history PR is landed.

File: service/frontend/operatorHandler.go [hunk 2]
```diff
@@ -0,0 +1,303 @@
+// The MIT License
```
Comment (alexshtin) [inline service/frontend/operatorHandler.go:1]: Go idiomatic name for this file is `operator_handler.go`. I know, we didn't follow this rule, but all new stuff we ads is following. One day I will rename other files.

File: service/frontend/operatorHandler.go [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+		metricsClient: args.MetricsClient,
+		saProvider:    args.SaProvider,
+		saManager:     args.SaManager,
+	}
+
+	return handler
+}
+
+// Start starts the handler
+func (oh *OperatorHandlerImpl) Start() {
```
Comment (alexshtin) [inline service/frontend/operatorHandler.go:96]:
```suggestion
func (h *OperatorHandlerImpl) Start() {
```
`oh` is too funny.

**Review by alexshtin — COMMENTED** [review]
Most of the handler code is copied from `AdminHandler`, right? And you gonna keep it there for some period of time. It would be great to reduce code duplication and have an instance of `OpearatorHandler` in `AdminHandeler` and call these 3 methods from it. With proper comment that it is deprecated and so on.

**Comment by Spikhalskiy** [comment]
@alexshtin I would generally agree with you about code duplication if there are two different active endpoints that are using the same code. 
Here the situation is a bit different, we will have a deprecated and a new version of the endpoints. I think we should go copy-paste route in this situation because it will SIMPLIFY support and REDUCE the burden, not increase them
For example, the new operator endpoint doesn't return some of the data that is needed by the deprecated versions. 

The way you propose to do by using operator_handler in admin_handler will not fly, because operator_handler doesn't provide a superset, the endpoints are slightly different.
Striving to remove code duplication here will require the creation of a shared manager that is used by an old and a new endpoint and provides a superset of functionality. 

This way, every time you will be editing the new version of the code you will have to worry about the deprecated version (while you don't need to edit and update them after we migrate tctl and UI).
If you go dedup way, you also can't just remove the deprecated endpoints. You will need to 1. remove unused functionality from the shared code 2. move the shared code to operator handler.

**Review by alexshtin — APPROVED** [review]

**Review by yiminc — APPROVED** [review]

**Pull request merged**